src/Init/Core.lean

```diff
@@ -502,29 +504,21 @@ class SDiff (α : Type u) where
   -/
   sdiff : α → α → α
 
-/-- Subset relation: `a ⊆ b`  -/
-infix:50 " ⊆ " => Subset
-
-/-- Strict subset relation: `a ⊂ b`  -/
-infix:50 " ⊂ " => SSubset
-
-/-- Superset relation: `a ⊇ b`  -/
-infix:50 " ⊇ " => Superset
-
-/-- Strict superset relation: `a ⊃ b`  -/
-infix:50 " ⊃ " => SSuperset
-
-/-- `a ∪ b` is the union of`a` and `b`. -/
-infixl:65 " ∪ " => Union.union
-
-/-- `a ∩ b` is the intersection of`a` and `b`. -/
-infixl:70 " ∩ " => Inter.inter
-
-/--
-`a \ b` is the set difference of `a` and `b`,
-consisting of all elements in `a` that are not in `b`.
--/
-infix:70 " \\ " => SDiff.sdiff
+@[inherit_doc] infix:50 " ⊆ " => Subset
+@[inherit_doc] infix:50 " ⊂ " => SSubset
+@[inherit_doc] infix:50 " ⊇ " => Superset
+@[inherit_doc] infix:50 " ⊃ " => SSuperset
+@[inherit_doc] infixl:65 " ∪ " => Union.union
+@[inherit_doc] infixl:70 " ∩ " => Inter.inter
+@[inherit_doc] infix:70 " \\ " => SDiff.sdiff
+
+macro_rules | `($x ⊆ $y) => `(binrel% Subset $x $y)
+macro_rules | `($x ⊂ $y)  => `(binrel% SSubset $x $y)
+macro_rules | `($x ⊇ $y)  => `(binrel% Superset $x $y)
+macro_rules | `($x ⊃ $y)  => `(binrel% SSuperset $x $y)
+macro_rules | `($x ∪ $y) => `(binop% Union.union $x $y)
+macro_rules | `($x ∩ $y)  => `(binop% Inter.inter $x $y)
+macro_rules | `($x \ $y)  => `(binop% SDiff.sdiff $x $y)
```

Comment:
For this PR, let's please not touch any of these operations, nor `HasEquiv.Equiv`. These make the PR much more complicated, and it's not clear to me if any of these should use the special elaborator. Sets are very different from arithmetic.

If we can stick with just `Ne` and `bne`, then we can get this in ASAP and it will make it into the next release.